### PR TITLE
Fix programme builder timezone issue #3960

### PR DESCRIPTION
**Pull request by roryc89**
addresses main app issue  #3960

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{
```
Comment (naazy) [inline .eslintrc]: :+1:

File: src/MainUpdate.elm [hunk 2]
```diff
@@ -604,8 +601,10 @@ update msg model =
             UpdateDates datesList ->
                 let
                     datesListToDateWithoutTime =
-                        datesList
-                            |> List.map DateUtils.valueStringToDateWithoutTime
+                        Debug.log "UpdateDates datesListToDateWithoutTime"
+                            (Debug.log "UpdateDates datesList" datesList
```
Comment (naazy) [inline src/MainUpdate.elm]: Logs can be removed

File: src/MainUpdate.elm [hunk 3]
```diff
@@ -637,8 +636,8 @@ update msg model =
                             datesWithSessionsWithUpdatedDates
                 in
                     ( { model
-                        | datesWithSessions = orderedDatesWithSessions
-                        , pickedDates = datesListToDateWithoutTime
+                        | datesWithSessions = Debug.log "orderedDatesWithSessions" orderedDatesWithSessions
+                        , pickedDates = Debug.log "datesListToDateWithoutTime" datesListToDateWithoutTime
```
Comment (naazy) [inline src/MainUpdate.elm]: Do we want these logs?

File: src/MainUpdate.elm [hunk 4]
```diff
@@ -658,7 +657,10 @@ update msg model =
             UpdatePickedDates pickedDatesList ->
                 let
                     dateWithoutTimeList =
-                        List.map DateUtils.valueStringToDateWithoutTime pickedDatesList
+                        Debug.log "dateWithoutTimeList"
+                            (List.map DateUtils.pikadayValueToDate
+                                (Debug.log "pickedDatesList" pickedDatesList)
+                            )
```
Comment (naazy) [inline src/MainUpdate.elm]: again do we want these logs?

File: src/TableView.elm [hunk 5]
```diff
@@ -123,17 +126,19 @@ viewDateCell dateWithSessions timeDelimiters firstTime =
                         span [ class "prog-table__time " ] [ text t ]
                     )
 
-        elmDate =
-            DateUtils.dateWithoutTimeToDate dateWithSessions.date
+        -- elmDate =
+        --     DateUtils.dateWithoutTimeToDate dateWithSessions.date
```
Comment (naazy) [inline src/TableView.elm]: remove please

File: src/DateUtils.elm [hunk 6]
```diff
@@ -3,22 +3,27 @@ module DateUtils
...
         , timeOfDayToTime
         , valueStringToDateWithoutTime
+        , pikadayValueToDate
         )
 
 import Date
 import MainModel exposing (..)
 import Time
 import List.Extra
+import Date.Extra exposing (calendarDate)
```
Comment (naazy) [inline src/DateUtils.elm]: 🎉

File: src/DateUtils.elm [hunk 7]
```diff
@@ -201,3 +330,11 @@ timeIsBetween min max t =
                     t1.minute >= t2.minute
     in
         (greaterThanOrEqual t min) && (greaterThanOrEqual max t)
+
+
+dateWithoutTimeToDay : DateWithoutTime -> String
+dateWithoutTimeToDay d =
+    Date.Extra.calendarDate d.year (intToMonth d.month) d.day
+        |> Date.Extra.fromSpec Date.Extra.local Date.Extra.noTime
```
Comment (naazy) [inline src/DateUtils.elm]: ❤️

File: src/DateUtils.elm [hunk 8]
```diff
@@ -94,11 +99,47 @@ valueStringToDateWithoutTime dateString =
...
+                |> Result.map
+                    (\i ->
+                        if i < 2000 then
+                            i + 2000
+                        else
+                            i
+                    )
+                |> Result.withDefault -1
+
+        -- |> (+) 2000
```
Comment (SimonLab) [inline src/DateUtils.elm]: do you need this commented line?

**Pull request merged**